test/errorhandling.js

```diff
@@ -682,7 +682,7 @@ test("peeking inside autorun doesn't bork (global) state", t => {
         t.equal(b.diffValue, 0)
         t.equal(b.lowestObserverState, 0)
         t.equal(b.unboundDepsCount, 1)
-        t.equal(b.value, undefined)
+        t.notEqual(b.value, 3)
```

Comment:
I'm not particularly happy with this assertion. I wanted to write something like `t.ok(isCaughtException(b.value))` but as far as I can tell neither `CaughtException` nor `isCaughtException` are exported from `mobx.ts`. I considered exporting them from `mobx.ts`, but that's changing public API. Also considered importing them from `core/derivation.ts` directly, but that's inconsistent with all other tests.

This feels like it's still achieving the goal of the assertion (ensuring that the cached value is cleared) while avoiding those problems.

I'm open to suggestions of other approaches.